### PR TITLE
Use Google Cloud Storage as CDN for list

### PR DESCRIPTION
**Pull request by eddiemonge**
This turns this more into a cron job type thing that compiles the list
and then pushes it to Google Cloud Storage so sites/3rd parties/yo-cli
can pull it from there.

I didn't remove the server stuff yet so if we wanted to, we could run this side by side with the existing solutions.

@stephenplusplus since you are helping with gcloud, want to take a peek at this?

### CODE REVIEW

**Comment by SBoudrias** [comment]
Should we push this to a branch until we're sure of the technique?

**Comment by eddiemonge** [comment]
We could. Since I didn't modify anything in server.js there isn't much harm in merging it to master either. I've been testing it and it seems fine. I have the list up at https://storage.googleapis.com/yeoman-generator-list/cache.json but I don't have anything updating it at the moment. It was updated from my dev testing. If its fine then I'll push it up to somewhere so it can be continuously updated.

**Comment by SBoudrias** [comment]
LGTM then!

**Comment by addyosmani** [comment]
Very late LGTM2.

File: index.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          'ETag': Plugins.getETag()
+        }
+      }
+    }))
+    .on('error', function(err) {
+      log.error('Error publishing the list: ', err);
+    })
+    .on('finish', function () {
+      log.info('Pushed cache to CDN');
+      file.makePublic(function (err) {
```
Comment (stephenplusplus) [inline index.js]: @eddiemonge you can skip this step by setting the default object ACL to public: https://cloud.google.com/storage/docs/access-control?hl=en#changingdefaultobject

```sh
$ gsutil defacl ch -u AllUsers:R gs://yeoman-generator-list
```

File: index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+var Plugins = new pluginCache(npmListKeyword, apiLimit);
+var update = function () {
+  Plugins.update().finally(function () {
+    var file = bucket.file('cache.json');
+    Plugins.getCacheStream()
+    .pipe(file.createWriteStream({
+      gzip: true,
+      metadata: {
+        contentType: 'application/json',
+        cacheControl: 'max-age=3600',
```
Comment (stephenplusplus) [inline index.js]: This is the default for public files, according to https://cloud.google.com/storage/docs/gsutil/commands/defacl

> By default, publicly readable objects are served with a Cache-Control header allowing such objects to be cached for 3600 seconds.

**Pull request merged**